algorithms/indexing/indexer.py

```diff
@@ -705,7 +705,18 @@ def index(self):
                 rotation_matrix_differences(self.refined_experiments.crystals())
             )
 
-        self._xyzcal_mm_to_px(self.experiments, self.refined_reflections)
+        # in case refinement failed, clean the reflections which may have had
+        # indices assigned to an experiment which was not saved...
+        refl = self.refined_reflections
+        expt = self.refined_experiments
```

Comment:
I would prefer to just use the original variables as they are, without re-assigning to a shorter name for a few lines of code.

Reply:
Acknowledged

However the full variable names did make the code rather harder to read since the lines become _very_ long. 

I'll have a look while fixing this.